### DISCUSSION TITLE
Database per profile

### DISCUSSION THREAD

**Discussion by becjit** [simolus3/drift#2596] [Q&A] [answered]
Hi,
  My app needs a separate database per profile. However the recommendation is to create a singleton. If I try to instantiate two Database objects using separate db files. I get the following error

 

> When these two databases use the same QueryExecutor, race conditions will occur and might corrupt the database.
> flutter: Try to follow the advice at https://drift.simonbinder.eu/faq/#using-the-database or, if you know what you're doing, set driftRuntimeOptions.dontWarnAboutMultipleDatabases = true

What is the recommended way of getting one db per user?
BTW, for all the DBs the schema remains the same

**Comment by simolus3** [comment, marked as answer]
If all those databases are using different paths that are user-specific, you can set `driftRuntimeOptions.dontWarnAboutMultipleDatabases = true` and make the error go away. The warning occurs because sometimes, people accidentally create multiple databases pointing towards the same underlying file - and that can cause issues with migrations if not done carefully.

**Reply by becjit** [reply]
@simolus3 thanks so much. So as long as I maintain two db files things should be ok? Any other thing I should be aware of?

**Reply by simolus3** [reply]
No other things come to mind as long as the two databases use different files :)
